Modules/Test/classes/class.ilTestEvaluationGUI.php

```diff
@@ -1145,7 +1145,15 @@ function outUserResultsOverview()
 		}
 		else
 		{
-			$template->setCurrentBlock("pass_overview");
+			//gev-patch start #2394
+                        if (((array_key_exists("pass", $_GET)) && (strlen($_GET["pass"]) > 0)) || (!is_null($pass))) {
+                            $pass = $_GET["pass"];
+                        } else {
+                            $pass = $this->object->_getResultPass($active_id);
+                        }
+                        //gev-patch end
```

Comment:
I think this is the wrong position to set $pass. It is never used in this else block. Maybe you look where the detail form will be rendered an set $pass just there. If you do this way, you can delete some copied stuff.

Questions? Just ask!